newIDE/app/src/AssetStore/NewObjectFromScratch.js

```diff
@@ -0,0 +1,397 @@
+          assetStorePackTag: 'resource bars',
+          requiredExtensions: [
+            {
+              extensionName: 'PanelSpriteContinuousBar',
+              extensionVersion: '1.0.0',
+            },
+          ],
+        },
+        {
+          name: 'TiledUnitsBar::TiledUnitsBar',
+          fullName: i18n._(t`Resource bar (separated units)`),
+          description: i18n._(
+            t`A bar that represents a resource in the game (health, mana, ammo, etc).`
+          ),
+          iconFilename:
+            'data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48IURPQ1RZUEUgc3ZnIFBVQkxJQyAiLS8vVzNDLy9EVEQgU1ZHIDEuMS8vRU4iICJodHRwOi8vd3d3LnczLm9yZy9HcmFwaGljcy9TVkcvMS4xL0RURC9zdmcxMS5kdGQiPjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgdmVyc2lvbj0iMS4xIiBpZD0ibWRpLWRvdHMtaG9yaXpvbnRhbCIgd2lkdGg9IjI0IiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiPjxwYXRoIGQ9Ik0xNiwxMkEyLDIgMCAwLDEgMTgsMTBBMiwyIDAgMCwxIDIwLDEyQTIsMiAwIDAsMSAxOCwxNEEyLDIgMCAwLDEgMTYsMTJNMTAsMTJBMiwyIDAgMCwxIDEyLDEwQTIsMiAwIDAsMSAxNCwxMkEyLDIgMCAwLDEgMTIsMTRBMiwyIDAgMCwxIDEwLDEyTTQsMTJBMiwyIDAgMCwxIDYsMTBBMiwyIDAgMCwxIDgsMTJBMiwyIDAgMCwxIDYsMTRBMiwyIDAgMCwxIDQsMTJaIiAvPjwvc3ZnPg==',
+          assetStorePackTag: 'resource bars',
+          requiredExtensions: [
+            {
+              extensionName: 'TiledUnitsBar',
+              extensionVersion: '1.0.0',
+            },
+          ],
+        },
+        {
+          name: 'TextInput::TextInputObject',
+        },
+        {
+          name: 'Video::VideoObject',
+        },
+      ],
+      [translateExtensionCategory('Visual effect', i18n)]: [
+        {
+          name: 'Lighting::LightObject',
+        },
+        {
+          name: 'ParticleSystem::ParticleEmitter',
+        },
+      ],
+      [translateExtensionCategory('Advanced', i18n)]: [
+        {
+          name: 'TileMap::TileMap',
+        },
+        {
+          name: 'TileMap::CollisionMask',
+        },
+        {
+          name: 'PrimitiveDrawing::Drawer',
+        },
+      ],
+    };
+    return enumeratedObjectMetadatasByCategory;
+  };
+
+  const enumeratedObjectMetadatasByCategory: {
+    [string]: Array<EnumeratedObjectMetadata>,
+  } = React.useMemo(
+    () => {
+      const enumeratedObjectMetadatasByCategory = getDefaultEnumeratedObjectMetadatasByCategory(
```

Comment:
Maybe move this into a function `mergeInstalledWithDefaultEnumeratedObjectMetadataByCategory`

Reply:
Function outside the component if possible